ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -2774,6 +3107,16 @@ static vk_pipeline ggml_vk_guess_matmul_pipeline(ggml_backend_vk_context * ctx,
         break;
     }
 
+    if (ctx->device->coopmat2) {
+        if ((m % mmp->l->wg_denoms[0]) == 0 && (n % mmp->l->wg_denoms[1]) == 0) {
+            return aligned ? mmp->a_l : mmp->l;
+        }
+        if ((m % mmp->m->wg_denoms[0]) == 0 && (n % mmp->m->wg_denoms[1]) == 0) {
+            return aligned ? mmp->a_m : mmp->m;
+        }
+        return aligned ? mmp->a_s : mmp->s;
```

Comment:
Interesting. I guess the logic here is to only use larger tiles if this doesn't require padding? Do you think this is applicable to the generic shader too, or specific to your cm2 version?

Reply:
Right, it's trying to use the largest size that doesn't require bounds-checking, because the bounds-checking is moderately expensive. In the generic shader I think the bounds checking is a much smaller impact and is currently always enabled, so similar logic probably wouldn't be helpful without also adding non-bounds-checking variants.

Reply:
No, bounds checking only happens on the non-aligned f16 and f32 loads. For the vectorized and quant loads no checks are done.

Reply:
No, you're right, I was mistaken and though there were bounds checks on the loads. Since you're already getting the benefit of not bounds checking, then this logic probably wouldn't help for the existing shaders.